### PR TITLE
Reworked how dynamic processes are managed.

### PR DESCRIPTION
**Pull request by bbalser**

### CODE REVIEW

File: lib/elsa/dynamic_process_manager.ex [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    |> Enum.each(fn child ->
+      output = DynamicSupervisor.start_child(state.dynamic_supervisor, child)
+
+      Logger.debug(fn ->
+        "#{__MODULE__}: output of starting #{inspect(child)}: #{inspect(output)}"
+      end)
+    end)
+  end
+
+  defp run_initializer(initializer) do
```
Comment (jeffgrunewald) [inline lib/elsa/dynamic_process_manager.ex]: maybe a guard here for when initializer is a function/0 vs. an empty list?

File: lib/elsa/dynamic_process_manager.ex [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  catch
+    :exit, e ->
+      Logger.warn(fn -> "#{__MODULE__}: initializer raised exception, retrying: #{inspect(e)}" end)
+
+      Process.sleep(1_000)
+      run_initializer(initializer)
+  end
+
+  defp whereis(name) when is_atom(name) do
+    Process.whereis(name)
```
Comment (jeffgrunewald) [inline lib/elsa/dynamic_process_manager.ex]: if you're going to pull this into a private function here (just to add the atom guard on the supervisor name?) would it make sense to just fold in the `Process.monitor/1` and return the ref directly? Conversely, would it be better to bubble up higher to the public API that you're expecting to only accept an atom (module name) for your supervisor?

File: lib/elsa/consumer/worker/initializer.ex [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  defp child_spec(registry, topic, partition, init_arg) do
+    name = :"topic_consumer_worker_#{topic}_#{partition}"
+
+    {Elsa.Consumer.Worker,
+     init_arg
+     |> Keyword.put(:name, {:via, Elsa.Registry, {registry, name}})
+     |> Keyword.put(:topic, topic)
+     |> Keyword.put(:partition, partition)
+     |> IO.inspect()}
```
Comment (jeffgrunewald) [inline lib/elsa/consumer/worker/initializer.ex]: IO.inspect to remove

File: lib/elsa/group/supervisor.ex [hunk 4]
```diff
@@ -22,10 +30,13 @@ defmodule Elsa.Group.Supervisor do
     registry = registry(connection)
     group = Keyword.fetch!(init_arg, :group)
 
+    group_consumer_supervisor = {:via, Elsa.Registry, {registry, :group_consumer_supervisor}}
+
     children =
       [
         {DynamicSupervisor, [strategy: :one_for_one, name: {:via, Elsa.Registry, {registry, :worker_supervisor}}]},
-        consumer_supervisors(registry, topics, config),
+        {DynamicSupervisor, [strategy: :one_for_one, name: group_consumer_supervisor]},
```
Comment (jeffgrunewald) [inline lib/elsa/group/supervisor.ex]: if the workers need an initializer, do they also need to be started under a dynamic_process_manager? Or do workers under a group-based consumer not need initializers?
Also, is there a need to have a separate dynamic supervisor for the workers under a group or can they share a dynamic supervisor with the group consumer?

File: lib/elsa/producer.ex [hunk 5]
```diff
@@ -35,6 +35,7 @@ defmodule Elsa.Producer do
     case Process.whereis(registry) do
       nil ->
         {:ok, pid} = Elsa.Supervisor.start_link(endpoints: endpoints, connection: connection, producer: [topic: topic])
+        ready?(connection)
```
Comment (jeffgrunewald) [inline lib/elsa/producer.ex]: does this case statement want to turn into a nested `with` block not that you have these conditions that want to execute in order and only conditionally when the previous ones succeed?

File: lib/elsa/group/supervisor.ex (hunk 4 above)
Comment (bbalser) [reply lib/elsa/group/supervisor.ex]: Going to look into this, made a separate for the moment just to avoid any conflict.

**Pull request merged**